disco/disco.go

```diff
@@ -204,7 +215,7 @@ func (d *Disco) discover(hostname svchost.Hostname) (*Host, error) {
 
 	resp, err := client.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to request discovery document: %v", err)
+		return nil, ErrServiceDiscoveryNetworkRequest{err}
```

Comment:
I wonder if we can trigger this error in the unit tests ([maybe one of these? the tests are not specific](https://github.com/hashicorp/terraform-svchost/blob/master/disco/disco_test.go#L179-L204)) and then assert that err is ErrServiceDiscoveryNetworkRequest